### ISSUE TITLE
CKV_SECRET_6 cannot be suppressed other than using comments which are not carried over in "aws cloudformation package" command

### ISSUE THREAD

**Issue by andyfase** [bridgecrewio/checkov#5282]
**Describe the issue**

Secret Checks seem to require a "higher" level of suppression than any other rule. From https://www.checkov.io/2.Basics/Suppressing%20and%20Skipping%20Policies.html

"To suppress secrets checks in any configuration file a comment needs to be added directly before, after or next to the infringing line."

Therefore secret checks suppression is limited to comments only. 

In our CI/CD pipeline CloudFormation templates are packages using `aws cloudformation package` command. This packaging ingests the base YAML, converts whats needed and then outputs a "built" YAML file. However this process removes comments from the original YAML file  - hence removes the suppression rules. To get around this we typically use Metadata style suppression i.e. 

```
  FargateTaskExecutionServiceRole:
    Metadata:
      checkov:
        skip:
          - id: CKV_SECRET_6
            comment: "False Positive ignored"
```

Clearly this doesn't suppress  and therefore I'm unable to use `checkov` on a packaged CFN. Its unclear to me why secret suppression would be limited in this way, and I ask for that decision to be re-evaluated, if even just for the support of CloudFormation

**Examples**
original template (which actually produces a false positive but that is by the by)
```
AWSTemplateFormatVersion: '2010-09-09'

Parameters:
  DatabaseSecretArn:
    Type: AWS::SSM::Parameter::Value<String>

Resources:
  Role:
    Metadata:
      checkov:
        skip:
          - id: CKV_SECRET_6
            comment: "This Doesnt Work"
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
        - Effect: Allow
          Principal:
            Service: 
              - service.amazonaws.com
          Action:
            - sts:AssumeRole
      Policies:
        - PolicyName: GetDatabaseSecret
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: "Allow"
                Action:
                  - "secretsmanager:GetSecretValue" 
                Resource: !Ref DatabaseSecretArn

```

**Version (please complete the following information):**
 - Checkov Version [e.g. 22]
2.3.115

**Additional context**
The AWS CLI uses the underlying pyyaml python library for YAML import/export - https://github.com/aws/aws-cli/blob/develop/awscli/customizations/cloudformation/yamlhelper.py

This library does not support comment preservation unfortunately.

**Comment by andyfase** [comment]
Note for anyone facing something similiar it is possible to trick checkov to bypass the check with something like:

```
Parameters:
  DatabaseSecretArn:
    Type: AWS::SSM::Parameter::Value<String>
  ## See https://github.com/bridgecrewio/checkov/issues/5282
  ## This stops checkov seeing this is a secret in the first place
  CheckOVSecretBypass:
    Type: String
    Default: "GetSecretValue"

.....
      Policies:
        - PolicyName: GetDatabaseSecret
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: "Allow"
                Action:
                  - !Sub "secretsmanager:${CheckOVSecretBypass}" 
                Resource: !Ref DatabaseSecretArn
```

**Comment by gruebel** [comment]
hey @andyfase thanks for reaching out.

`CKV_SECRET_6` is probabilistic secrets scan, which means it is more likely to have false-positives. That being said, most of our secrets scanning is decoupled from IaC scanning, so if a secret is found it doesn't has any context.

There are currently no plans on supporting what you are asking for. I keep it is a feature request, but don't expect anything in the next months.
